src/message.rs

```diff
@@ -24,6 +24,8 @@ use crate::{
     header::{self, HeaderMap},
 };
 
+/// A predicate used to filter messages.
+pub(crate) type MessageFilter = Box<dyn Fn(&Message) -> bool + Send + Sync>;
```

Comment:
Might want to explore another name to avoid FilterSubject option/property on consumers.

Reply:
Went with message preprocessor.